packages/react-router/src/link.tsx

```diff
@@ -106,7 +106,7 @@ export function useLinkProps<
     structuralSharing: true as any,
   })
 
-  const isRelativeFromPath = options.relative === 'path'
+  const isRelativeFromPath = options.unsafeRelative === 'path'
```

Comment:
@birkskyum reminder about porting this behaviour and its tests over to Solid.

It doesn't need to block this PR, but just something to be done after.